### ISSUE TITLE
Show submodule recipes in --choose

### ISSUE THREAD

**Issue by casey** [casey/just#2110]
From [this comment](https://github.com/casey/just/issues/929#issuecomment-2136874285)

@foresterre reports that submodule recipes aren't appearing in `--choose`

I wasn't actually able to reproduce this. Doing this:

```
$ cat test.just
mod foo
$ cat foo.just
bar:
$ cargo run -- --unstable --justfile test.just --choose
```

Showed `foo::bar` in my chooser.

**Comment by foresterre** [comment]
Reproduced this on both windows and ubuntu using: https://github.com/foresterre/sic/blob/960375b1fd37960e8cc8d4a22b1fa1ecbde32a25/justfile#L17

In this repo (branch 'avif-decoder', see commit hash above), I call just from the root of the repo with `just --unstable`. The result looks like this on ubuntu:

![image](https://github.com/casey/just/assets/5955761/5edc487b-4879-4002-8d66-ffb6e340dfcc)

and on windows: https://github.com/casey/just/issues/929#issuecomment-2136874285 (see earlier) comment

**Top level justfile from [sic](https://github.com/foresterre/sic/blob/960375b1fd37960e8cc8d4a22b1fa1ecbde32a25/justfile) project above for reference**

<details>
<summary>sic top level justfile </summary>

```
# NB: mod requires Just >= 1.19 and just --unstable
mod clippy  '.justfiles/clippy.just'
mod deny    '.justfiles/deny.just'
mod fmt     '.justfiles/fmt.just'
mod msrv    '.justfiles/msrv.just'
mod test    '.justfiles/test.just'
mod dav1d   '.justfiles/dav1d.just'

[windows]
default:
    @echo 'On Windows, run just using:'
    @echo 'just --unstable --shell pwsh.exe --shell-arg -c'
    @just --choose --unstable

[unix]
default:
    @just --choose --unstable

before-push:
    # do fmt
    just --unstable fmt
    # run checks
    just --unstable fmt check
    just --unstable dav1d
    just --unstable clippy
    just --unstable test
    just --unstable msrv
    just --unstable deny
```

</details>

---

Haven't tried latest version from git yet;

I used (ubuntu, installed via cargo install just):
```
> just --version
just 1.25.0
```

**Comment by foresterre** [comment]
Installed latest release on ubuntu with `cargo install just` (1.27.0), this solves the issue indeed. My apologies:

![image](https://github.com/casey/just/assets/5955761/4c0502e1-6c41-46eb-b7d3-343674c19941)

However, it does appear that the chooser has a different issue, namely, showing the recipe content:

![image](https://github.com/casey/just/assets/5955761/fb78cc74-f1a9-4083-9453-2fce3d31db0d)

This is the case for all recipes which are from the modules. I can select and run them via the chooser just fine, despite it stating that the recipe is unknown.

Do you want a new issue on github for this (so the reference to the issue from which this one was split stays intact) ?

**Comment by foresterre** [comment]
For completeness sake, and since I was installing just from source anyways and this was hardly any extra effort, I also tested this last one also with the latest git [77a6e02](https://github.com/casey/just/commit/77a6e02964f2f7607bd5f6163de5df5a1c13661d), where this last issue is still present:

![image](https://github.com/casey/just/assets/5955761/31a5c4a2-975a-48eb-baa4-65163a2ca605)


i.e. `cargo install just --git https://github.com/casey/just.git --rev 77a6e02964f2f7607bd5f6163de5df5a1c13661d`

**Comment by casey** [comment]
Nice, glad it works! I just merged #2111, which makes `--show` take a recipe path, so recipe preview when choosing should work now.

**Comment by casey** [comment]
This is still broken. Recipes are printed out with space-separated paths, but passed as a single argument to the `--show` preview command, which `--show` doesn't handle.

**Comment by casey** [comment]
Fixed, hopefully this time for good, in #2141.
